contracts/BatchReveal.sol

```diff
@@ -88,24 +88,126 @@ abstract contract BatchReveal is
     /// @param batchSeed The random number drawn
     event Reveal(uint256 batchNumber, uint256 batchSeed);
 
+    /// @dev Emitted on setRevealBatchSize()
+    /// @param revealBatchSize New reveal batch size
+    event RevealBatchSizeSet(uint256 revealBatchSize);
+
+    /// @dev Emitted on setRevealStartTime()
+    /// @param revealStartTime New reveal start time
+    event RevealStartTimeSet(uint256 revealStartTime);
+
+    /// @dev Emitted on setRevealInterval()
+    /// @param revealInterval New reveal interval
+    event RevealIntervalSet(uint256 revealInterval);
+
+    modifier initialized() {
+        if (!isInitialized()) {
+            revert Launchpeg__BatchRevealNotInitialized();
+        }
+        _;
+    }
+
+    modifier revealNotStarted() {
+        if (lastTokenRevealed != 0) {
+            revert Launchpeg__BatchRevealStarted();
+        }
+        _;
+    }
+
     /// @dev BatchReveal initialization
-    /// @param _revealBatchSize Size of the batch reveal
+    /// @param _revealBatchSize Size of the batch reveal. Set to 0 to disable batch reveal.
     /// @param _collectionSize Needs to be sent by child contract
+    /// @param _revealStartTime Batch reveal start time
+    /// @param _revealInterval Batch reveal interval
     function initializeBatchReveal(
         uint256 _revealBatchSize,
-        uint256 _collectionSize
+        uint256 _collectionSize,
+        uint256 _revealStartTime,
+        uint256 _revealInterval
     ) internal onlyInitializing {
+        if (_collectionSize == 0) {
```

Comment:
i notice this is not checked in the base contract - should we do so?

Reply:
Hm if we want to verify that, I think we should do it in `initializeBaseLaunchpeg` no?

Reply:
yes agreed